airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,190 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import requests
+
+from airflow import __version__
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+from requests import exceptions as requests_exceptions
+
+
+try:
+    from urllib import parse as urlparse
+except ImportError:
+    import urlparse
+
+
+SUBMIT_RUN_ENDPOINT = ('POST', 'api/2.0/jobs/runs/submit')
+GET_RUN_ENDPOINT = ('GET', 'api/2.0/jobs/runs/get')
+USER_AGENT_HEADER = {'user-agent': 'airflow-{v}'.format(v=__version__)}
+
+
+class DatabricksHook(BaseHook):
+    """
+    Interact with Databricks.
+    """
+    def __init__(
+            self,
+            databricks_conn_id='databricks_default',
+            timeout_seconds=180,
+            retry_limit=3):
+        """
+        :param databricks_conn_id: The name of the databricks connection to use.
+        :type databricks_conn_id: string
+        :param timeout_seconds: The amount of time in seconds the requests library
+            will wait before timing-out.
+        :type timeout_seconds: int
+        :param retry_limit: The number of times to retry the connection in case of
+            service outages.
+        :type retry_limit: int
+        """
+        self.databricks_conn_id = databricks_conn_id
+        self.databricks_conn = self.get_connection(databricks_conn_id)
+        self.timeout_seconds = timeout_seconds
+        self.retry_limit = retry_limit
+
+    def _parse_host(self, host):
+        """
+        The purpose of this function is to be robust to improper connections
+        settings provided by users, specifically in the host field.
+
+        For example -- when users supply ``https://XX.cloud.databricks.com`` as the
+        host, we must strip out the protocol to get the host.
+            _parse_host(https://XX.cloud.databricks.com) -> XX.cloud.databricks.com.
+
+        In the case where users supply the correct ``XX.cloud.databricks.com`` as the
+        host, this function is a no-op.
+            _parse_host(XX.cloud.databricks.com) -> XX.cloud.databricks.com.
+        """
+        urlparse_host = urlparse.urlparse(host).hostname
+        if urlparse_host:
+            # In this case, host = https://XX.cloud.databricks.com
+            return urlparse_host
+        else:
+            # In this case, host = XX.cloud.databricks.com
+            return host
+
+    def _do_api_call(self, endpoint_info, api_parameters):
+        """
+        Utility function to perform an API call with retries
+        :param endpoint_info: Tuple of method and endpoint
+        :type endpoint_info: (string, string)
+        :param api_parameters: Parameters for this API call.
```

Comment:
As discussed https://github.com/apache/incubator-airflow/pull/2202#discussion_r109754570. I'm going to change to `json`.